#Md Raihanul Islam Bhuiyan
###Module-7: Pandas Data Analysis Using Heart Disease Dataset

In [ ]:
import pandas as pd

### Question: 1
Dataset Loading and Initial Exploration

In [ ]:
import kagglehub
path = kagglehub.dataset_download("johnsmith88/heart-disease-dataset")

Using Colab cache for faster access to the 'heart-disease-dataset' dataset.


In [ ]:
import os
data = pd.read_csv(os.path.join(path, 'heart.csv'))

In [ ]:
data

,age,sex,cp,trestbps,chol,fbs,restecg,thalach,exang,oldpeak,slope,ca,thal,target
0,52,1,0,125,212,0,1,168,0,1.0,2,2,3,0
1,53,1,0,140,203,1,0,155,1,3.1,0,0,3,0
2,70,1,0,145,174,0,1,125,1,2.6,0,0,3,0
3,61,1,0,148,203,0,1,161,0,0.0,2,1,3,0
4,62,0,0,138,294,1,1,106,0,1.9,1,3,2,0
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
1020,59,1,1,140,221,0,1,164,1,0.0,2,0,2,1
1021,60,1,0,125,258,0,0,141,1,2.8,1,1,3,0
1022,47,1,0,110,275,0,0,118,1,1.0,1,1,2,0
1023,50,0,0,110,254,0,0,159,0,0.0,2,0,2,1


In [ ]:
# Displaying first 5 rows

data.head(5)


,age,sex,cp,trestbps,chol,fbs,restecg,thalach,exang,oldpeak,slope,ca,thal,target
0,52,1,0,125,212,0,1,168,0,1.0,2,2,3,0
1,53,1,0,140,203,1,0,155,1,3.1,0,0,3,0
2,70,1,0,145,174,0,1,125,1,2.6,0,0,3,0
3,61,1,0,148,203,0,1,161,0,0.0,2,1,3,0
4,62,0,0,138,294,1,1,106,0,1.9,1,3,2,0


In [ ]:
#Displaying last 5 rows

data.tail(5)

,age,sex,cp,trestbps,chol,fbs,restecg,thalach,exang,oldpeak,slope,ca,thal,target
1020,59,1,1,140,221,0,1,164,1,0.0,2,0,2,1
1021,60,1,0,125,258,0,0,141,1,2.8,1,1,3,0
1022,47,1,0,110,275,0,0,118,1,1.0,1,1,2,0
1023,50,0,0,110,254,0,0,159,0,0.0,2,0,2,1
1024,54,1,0,120,188,0,1,113,0,1.4,1,1,3,0


In [ ]:
# Rows : Columns

print("Shape:", data.shape)


Shape: (1025, 14)


### Question: 2
Dataset Structure and Missing Value Analysis

In [ ]:
# Displaying all the column names

data.columns

Index(['age', 'sex', 'cp', 'trestbps', 'chol', 'fbs', 'restecg', 'thalach',
       'exang', 'oldpeak', 'slope', 'ca', 'thal', 'target'],
      dtype='object')

In [ ]:
# Showing dataset information including data types and non-null counts

data.info()


<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1025 entries, 0 to 1024
Data columns (total 14 columns):
 #   Column    Non-Null Count  Dtype  
---  ------    --------------  -----  
 0   age       1025 non-null   int64  
 1   sex       1025 non-null   int64  
 2   cp        1025 non-null   int64  
 3   trestbps  1025 non-null   int64  
 4   chol      1025 non-null   int64  
 5   fbs       1025 non-null   int64  
 6   restecg   1025 non-null   int64  
 7   thalach   1025 non-null   int64  
 8   exang     1025 non-null   int64  
 9   oldpeak   1025 non-null   float64
 10  slope     1025 non-null   int64  
 11  ca        1025 non-null   int64  
 12  thal      1025 non-null   int64  
 13  target    1025 non-null   int64  
dtypes: float64(1), int64(13)
memory usage: 112.2 KB


In [ ]:
# Identifying missing values in each column

data.isna().sum()

,0
age,0
sex,0
cp,0
trestbps,0
chol,0
fbs,0
restecg,0
thalach,0
exang,0
oldpeak,0


### Question: 3
Categorical Data Analysis

In [ ]:
# Patient with heart disease = 1, Patient without heart disease = 0

data['target'].value_counts()

,count
target,
1,526
0,499


In [ ]:
# 1 = Male, 2 = Female

data['sex'].value_counts()

,count
sex,
1,713
0,312


### Question: 4
Data Selection Using iloc

In [ ]:
# First 10 rows of the dataset

data.iloc[0:10]

,age,sex,cp,trestbps,chol,fbs,restecg,thalach,exang,oldpeak,slope,ca,thal,target
0,52,1,0,125,212,0,1,168,0,1.0,2,2,3,0
1,53,1,0,140,203,1,0,155,1,3.1,0,0,3,0
2,70,1,0,145,174,0,1,125,1,2.6,0,0,3,0
3,61,1,0,148,203,0,1,161,0,0.0,2,1,3,0
4,62,0,0,138,294,1,1,106,0,1.9,1,3,2,0
5,58,0,0,100,248,0,0,122,0,1.0,1,0,2,1
6,58,1,0,114,318,0,2,140,0,4.4,0,3,1,0
7,55,1,0,160,289,0,0,145,1,0.8,1,1,3,0
8,46,1,0,120,249,0,0,144,0,0.8,2,0,3,0
9,54,1,0,122,286,0,0,116,1,3.2,1,2,2,0


In [ ]:
# Extracting columns representing age, sex, and cholesterol

data.iloc[:, [1, 2, 4]]

,sex,cp,chol
0,1,0,212
1,1,0,203
2,1,0,174
3,1,0,203
4,0,0,294
...,...,...,...
1020,1,1,221
1021,1,0,258
1022,1,0,275
1023,0,0,254


In [ ]:
# Selecting rows (20 - 30) and columns  (0 - 4)

data.iloc[20:31, 0:5]

,age,sex,cp,trestbps,chol
20,60,1,2,140,185
21,67,0,0,106,223
22,45,1,0,104,208
23,63,0,2,135,252
24,42,0,2,120,209
25,61,0,0,145,307
26,44,1,2,130,233
27,58,0,1,136,319
28,56,1,2,130,256
29,55,0,0,180,327


### Question:5
Data Filtering Using Conditions

In [ ]:
# Patients older than 50 years

data[data['age'] > 50]

,age,sex,cp,trestbps,chol,fbs,restecg,thalach,exang,oldpeak,slope,ca,thal,target
0,52,1,0,125,212,0,1,168,0,1.0,2,2,3,0
1,53,1,0,140,203,1,0,155,1,3.1,0,0,3,0
2,70,1,0,145,174,0,1,125,1,2.6,0,0,3,0
3,61,1,0,148,203,0,1,161,0,0.0,2,1,3,0
4,62,0,0,138,294,1,1,106,0,1.9,1,3,2,0
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
1016,65,1,3,138,282,1,0,174,0,1.4,1,1,2,0
1017,53,1,0,123,282,0,1,95,1,2.0,1,2,3,0
1020,59,1,1,140,221,0,1,164,1,0.0,2,0,2,1
1021,60,1,0,125,258,0,0,141,1,2.8,1,1,3,0


In [ ]:
# Patients who have heart disease and a cholesterol level greater than 240
data[(data["target"] == 1) & (data["chol"] > 240)]

,age,sex,cp,trestbps,chol,fbs,restecg,thalach,exang,oldpeak,slope,ca,thal,target
5,58,0,0,100,248,0,0,122,0,1.0,1,0,2,1
16,51,0,2,140,308,0,0,142,0,1.5,2,1,2,1
18,50,0,1,120,244,0,1,162,0,1.1,2,0,2,1
23,63,0,2,135,252,0,0,172,0,0.0,2,0,2,1
31,50,0,1,120,244,0,1,162,0,1.1,2,0,2,1
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
995,44,1,1,120,263,0,1,173,0,0.0,2,0,3,1
1004,51,0,2,140,308,0,0,142,0,1.5,2,1,2,1
1008,42,1,1,120,295,0,1,162,0,0.0,2,0,2,1
1011,45,1,1,128,308,0,0,170,0,0.0,2,0,2,1


In [ ]:
#.Count how many patients satisfy both conditions

count = data[(data["target"] == 1) & (data["chol"] > 240)].shape[0]

print(count)

222


### Findings:

1. This dataset contains 14 types of information on 1025 patients.

2. 13 types of data are in integer format and 1 (oldpeak) is in float or decimal format

3. The dataset has no missing value.

4. The number of patients with heart diseases is greater in the dataset.

5. This dataset mostly contains male data.

6. Most of the patients are over 50 years old.

7. Older people is likely to have more chilestorel in their blood